packages/vite/src/node/server/openBrowser.ts

```diff
@@ -68,17 +68,36 @@ function startBrowserProcess(browser: string | undefined, url: string) {
     process.platform === 'darwin' && (browser === '' || browser === OSX_CHROME)
 
   if (shouldTryOpenChromeWithAppleScript) {
-    try {
-      // Try our best to reuse existing tab
-      // on OS X Google Chrome with AppleScript
-      execSync('ps cax | grep "Google Chrome"')
-      execSync('osascript openChrome.applescript "' + encodeURI(url) + '"', {
-        cwd: join(VITE_PACKAGE_DIR, 'bin'),
-        stdio: 'ignore'
-      })
-      return true
-    } catch (err) {
-      // Ignore errors
+    // Will use the first open browser found from list
+    const supportedChromiumBrowsers = [
+      'Google Chrome Canary',
+      'Google Chrome Dev',
+      'Google Chrome Beta',
+      'Google Chrome',
+      'Microsoft Edge',
+      'Brave Browser',
+      'Vivaldi',
+      'Chromium'
```

Comment:
> The reason a custom applescript is used is to reuse the existing tab on the browser, since other browser would reuse by default. Does this happen for all chromium-based browsers?

As a Chromium user (e.g: Chrome Dev/ Chrome Beta/ Chrome Canary), I don't see those browsers reuse existing tab by default. It always spans a new tab whenever I start the dev server

https://user-images.githubusercontent.com/8603085/196241595-fbaa17e5-13cf-469e-b5ec-456f40145749.mov

> Also, should Google Chrome be put to the first of the list?

The current implementation opens only one browser (from browsers that are being opened at the moment) by priority in the list (the browser that appears earlier in the array has higher priority).

So, Google Chrome does not need to be the first on the list. As a web developer using Google Dev as a primary development browser, when I have both Chrome Dev and Chrome opened, I want the dev server open (and reuse the existing tab) on Chrome Dev.

If a user has only Google Chrome opened, only that browser is used to open a dev server (since the script only cares about opening browsers). Notice that `execSync('ps cax | grep "' + chromiumBrowser + '"');` is being used to check if that browser is being opened (It raised an error, which is ignored if that process is not existed).

> I'm also hoping that this longer list doesn't slow down the initial load for those who don't use Chrome.

It does not. Since we already have
```js
return true
```
after `execSync('osascript openChrome.applescript ...')` command, `startBrowserProcess` will check if the user has a particular browser opened by priority, then use it, then exit function.

**More context:**
- `packages/vite/src/node/server/openBrowser.ts` is heavily inspired by [this file](https://github.com/facebook/create-react-app/blob/d960b9e38c062584ff6cfb1a70e1512509a966e7/packages/react-dev-utils/openBrowser.js) almost 2 years ago. The original code updated to support other Chromium browsers as well. This PR is simply backporting that update.

Reply:
Thanks for the explanation! That all makes sense to me. And since cra has been using that for quite some time now, I think we can go ahead with this.

Reply:
Thanks for your review 🚀